prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -278,17 +281,22 @@ class Hl7Serializer(val metadata: Metadata) {
         report.schema.elements.forEach { element ->
             val value = report.getString(row, element.name) ?: return@forEach
 
+            if (suppressedFields.contains(element.hl7Field))
+                return@forEach
+
             if (element.hl7OutputFields != null) {
-                element.hl7OutputFields.forEach { hl7Field ->
+                element.hl7OutputFields.forEach outputFields@{ hl7Field ->
+                    if (suppressedFields.contains(hl7Field))
+                        return@outputFields
                     setComponent(terser, element, hl7Field, value)
                 }
-            } else if (element.hl7Field == "AOE" && element.type == Element.Type.NUMBER) {
+            } else if (element.hl7Field == "AOE" && element.type == Element.Type.NUMBER && !suppressAoe) {
```

Comment:
We might be suppressing the AOE questions, so we need to check that here as well.